pkg/remotedialer/session_manager.go

```diff
@@ -109,14 +112,21 @@ func (sm *sessionManager) add(clientKey string, conn *websocket.Conn, peer bool)
 }
 
 func (sm *sessionManager) remove(s *Session) {
+	var isPeer bool
 	sm.Lock()
 	defer sm.Unlock()
 
-	for _, store := range []map[string][]*Session{sm.clients, sm.peers} {
+	for i, store := range []map[string][]*Session{sm.clients, sm.peers} {
 		var newSessions []*Session
 
 		for _, v := range store[s.clientKey] {
 			if v.sessionKey == s.sessionKey {
+				if i == 0 {
+					isPeer = false
```

Comment:
How are you ensuring that that the first Session is always non-peer?

Reply:
This i is not index of sessions loop but it's index of store loop
```
119   for i, store := range []map[string][]*Session{sm.clients, sm.peers} {
```
That's why this sentence ensure first store is always non-peer. thus this sentence strongly depending on the order of store at line 119.